src/middlewared/middlewared/plugins/truecommand/portal.py

```diff
@@ -42,8 +42,14 @@ async def poll_api_for_status(self, job):
                 self.middleware.send_event(
                     'truecommand.config', 'CHANGED', fields=(await self.middleware.call('truecommand.config'))
                 )
-                await self.middleware.call('truecommand.dismiss_alerts')
+                if status['tc_state'] == 'running':
```

Comment:
We should get this key safely since we've already seen it not exist in some circumstances.